roadmap/implementers-guide/src/node/dispute/participation.md

```diff
@@ -69,3 +81,14 @@ Nothing to store
   1. includes `CandidateReceipts`
   1. includes `Vote`s
 1. Delete all relevant data associated with this dispute
+
+### Blacklisting bad block and invalid decendants
+
+In case a block was disputed successfully, and is now deemed invalid.
+
+1. Query `ChainApiMessage::Descendants` for all descendants of the disputed `Hash`.
+1. Blacklist all descendants of the disputed block `ChainApiMessage::Blacklist`.
+1. Check if the dispute block was finalized
+1. iff:
+    1. put the chain into governance mode
```

Comment:
I figure on the runtime, whenever a local dispute succeeds, it should put the chain into some governance-only mode. Should be as simple as disabling the `InclusionInherent` from that point onwards.

If it's an unfinalized chain, then the other logic described in this PR should just abandon that chain. If it's a finalized chain, then we're in governance mode for good, which is what we want.

Reply:
Anyway, that's a long way of saying "seems beyond the scope of this module".